featuretools/tests/primitive_tests/test_season.py

```diff
@@ -0,0 +1,64 @@
+from datetime import datetime
+
+import pandas as pd
+
+from featuretools.primitives import Season
+
+
+class TestSeason:
+    def test_regular(self):
+        primitive_instance = Season()
+        primitive_func = primitive_instance.get_function()
+        case = pd.date_range(start="2019-01", periods=12, freq="m").to_series()
+        answer = pd.Series(
+            [
+                "winter",
+                "winter",
+                "spring",
+                "spring",
+                "spring",
+                "summer",
+                "summer",
+                "summer",
+                "fall",
+                "fall",
+                "fall",
+                "winter",
+            ],
+            dtype="string",
+        )
+        given_answer = primitive_func(case)
+        pd.testing.assert_series_equal(
+            given_answer.reset_index(drop=True),
+            answer.reset_index(drop=True),
+        )
+
+    def test_nat(self):
+        primitive_instance = Season()
+        primitive_func = primitive_instance.get_function()
+        case = pd.Series(
+            [
+                "NaT",
+                "2019-02",
+                "2019-03",
+                "NaT",
+            ],
+        ).astype("datetime64")
+        answer = pd.Series([pd.NA, "winter", "winter", pd.NA], dtype="string")
+        given_answer = pd.Series(primitive_func(case))
+        pd.testing.assert_series_equal(given_answer, answer)
+
+    def test_datetime(self):
+        primitive_instance = Season()
+        primitive_func = primitive_instance.get_function()
+        case = pd.Series(
+            [
+                datetime(2011, 3, 1),
+                datetime(2011, 6, 1),
+                datetime(2011, 9, 1),
+                datetime(2011, 12, 1),
```

Comment:
Since we have special handling for leap year cases in the primitive, should we include a leap year/day in our testing as well?

Reply:
Fixed:
https://github.com/alteryx/featuretools/pull/2460/commits/8e29e97238b20c31aff1cc8b7f57f953d222367a